File: Go/Fruits into basket/904. Fruit Into Baskets.go
package main

import (
	"fmt"
)

func totalFruit(fruits []int) int {
	count := 0
	Maxcount := 0
	lastX := 1
	X := -1
	Y := -1
	for ndx, x := range fruits {
		if X == -1 {
			X = x
		}
		if Y == -1 && x != X {
			Y = x
		}
		if x != X && x != Y && X != -1 && Y != -1 {
			if Maxcount < count {
				Maxcount = count
			}
			count = ndx - lastX
			if fruits[lastX] == X {
				Y = x
			} else {
				X = x
			}
		}

		if ndx > 0 {
			if fruits[ndx] != fruits[ndx-1] {
				lastX = ndx
			}
		}
		count++
	}
	if Maxcount < count {
		Maxcount = count
	}
	return Maxcount
}
func main() {
	fmt.Println(totalFruit([]int{3, 3, 3, 1, 2, 1, 1, 2, 3, 3, 4}))
}

File: Python/904. Fruit Into Baskets.py
class Solution(object):
    def totalFruit(self, fruits):
        count,Maxcount,lastX=0,0,1
        basket,ndx=set(),0
        while(ndx<len(fruits)):
            if len(basket)<2:
                basket.add(fruits[ndx])
            elif not fruits[ndx] in basket:
                Maxcount=max(count,Maxcount)
                count=ndx-lastX
                basket={fruits[ndx-1],fruits[ndx]}
            if fruits[ndx]!=fruits[ndx-1]:
                lastX=ndx
            count+=1
            ndx+=1   
        return max(Maxcount,count)
if __name__ == "__main__":
    obj=Solution()
    Op=obj.totalFruit([1,2,3,2,2])
    print(Op)